src/monocle_apptrace/metamodel/spans/span_example.json

```diff
@@ -0,0 +1,140 @@
+{
+    "name": "llamaindex.retrieve",
+    "context": {
+        "trace_id": "0x93cd0bf865b3ffcc3cf9c075dc3e3797",
+        "span_id": "0x5d3f839e900bda24",
+        "trace_state": "[]"
+    },
+    "kind": "SpanKind.CLIENT",
+    "parent_id": "0x7a63d63e42ccac60",
+    "start_time": "2024-09-09T14:38:45.237182Z",
+    "end_time": "2024-09-09T14:38:45.620112Z",
+    "status": {
+        "status_code": "OK"
+    },
+    "attributes": {
+        "span.type": "Retrieval", 
+        "entity.count": 2,
+        "entity.1.name": "ChromaVectorStore",
+        "entity.1.type": "VectorDB.Chroma",
+        "entity.1.embedding-model-name": "BAAI/bge-small-en-v1.5",
+        "entity.2.name": "BAAI/bge-small-en-v1.5",
+        "entity.2.type": "Model.embedding",
+        "entity.2.model_name": "BAAI/bge-small-en-v1.5"
+    },
+    "events": [
+        {
+            "name": "data.input",
+            "timestamp": "timestamp",
+            "attributes": {
+                "context_input": "question: What is an americano?"
+            }
+        },
+        {
+            "name": "data.output",
+            "timestamp": "timestamp",
+            "attributes": {
+                "context_output": "Coffee is a hot drink made from the roasted and ground seeds (coffee beans) of a tropical shrub\nA latte consists of one or more shots of espresso, served in a glass (or sometimes a cup), into which hot steamed milk is added\nAmericano is a type of coffee drink prepared by diluting an espresso shot with hot water at a 1:3 to 1:4 ratio, resulting in a drink that retains the complex flavors of espresso, but in a lighter way"
+            }
+        }
+
+    ],
+    "links": [],
+    "resource": {
+        "attributes": {
+            "service.name": "coffee-bot"
+        },
+        "schema_url": ""
+    }
+},
+{
+    "name": "llamaindex.openai",
+    "context": {
+        "trace_id": "0x93cd0bf865b3ffcc3cf9c075dc3e3797",
+        "span_id": "0x8b6363e1937a4d7b",
+        "trace_state": "[]"
+    },
+    "kind": "SpanKind.CLIENT",
+    "parent_id": "0x7a63d63e42ccac60",
+    "start_time": "2024-09-09T14:38:45.622174Z",
+    "end_time": "2024-09-09T14:38:46.514120Z",
+    "status": {
+        "status_code": "OK"
+    },
+    "attributes": {
+        "span.type": "Inference",
+        "entity.count": 2,
+        "entity.1.name": "AzureOpenAI",
+        "entity.1.type": "Inference.Azure_oai",
+        "entity.1.provider_name": "openai.azure.com",
```

Comment:
what would be the difference between 
`entity.1.name` and `entity.1.provider_name` - can we not merge them  ?

Reply:
I believe provider name is an argument sent to Azure OpenAI inference API. Hence it's capturing here.

Reply:
provider_name is currently set to hostname of inference endpoint, so `openai.azure.com` in case of azopenai

ref: https://github.com/monocle2ai/monocle/blob/cc825cc6e91dadacea07b9cefeb4c4d688354557/src/monocle_apptrace/wrap_common.py#L187

cc @kshitiz-okahu

Reply:
provider_name is derived from the hostname of inference url for capturing it in span. It is not sent to openAI url.